File: src/main/java/org/apache/fineract/client/services/SelfClientApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.Call;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.FormDataBodyPart;
import org.apache.fineract.client.models.GetSelfClientsClientIdAccountsResponse;
import org.apache.fineract.client.models.GetSelfClientsClientIdChargesChargeIdResponse;
import org.apache.fineract.client.models.GetSelfClientsClientIdChargesResponse;
import org.apache.fineract.client.models.GetSelfClientsClientIdResponse;
import org.apache.fineract.client.models.GetSelfClientsClientIdTransactionsResponse;
import org.apache.fineract.client.models.GetSelfClientsClientIdTransactionsTransactionIdResponse;
import org.apache.fineract.client.models.GetSelfClientsResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface SelfClientApi {
  /**
   * 
   * 
   * @param clientId  (required)
   * @param contentLength  (optional)
   * @param file  (optional)
   * @return Call&lt;String&gt;
   */
  @retrofit2.http.Multipart
  @POST("self/clients/{clientId}/images")
  Call<String> addNewClientImage2(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Header("Content-Length") Long contentLength, @retrofit2.http.Part("file") FormDataBodyPart file
  );

  /**
   * 
   * 
   * @param clientId  (required)
   * @return Call&lt;String&gt;
   */
  @DELETE("self/clients/{clientId}/images")
  Call<String> deleteClientImage1(
    @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * List Clients associated to the user
   * The list capability of clients can support pagination and sorting.  Example Requests:  self/clients  self/clients?fields&#x3D;displayName,officeName  self/clients?offset&#x3D;10&amp;limit&#x3D;50  self/clients?orderBy&#x3D;displayName&amp;sortOrder&#x3D;DESC
   * @param displayName displayName (optional)
   * @param firstName firstName (optional)
   * @param lastName lastName (optional)
   * @param offset offset (optional)
   * @param status status (optional)
   * @param limit limit (optional)
   * @param orderBy orderBy (optional)
   * @param sortOrder sortOrder (optional)
   * @return Call&lt;GetSelfClientsResponse&gt;
   */
  @GET("self/clients")
  Call<GetSelfClientsResponse> retrieveAll36(
    @retrofit2.http.Query("displayName") String displayName, @retrofit2.http.Query("firstName") String firstName, @retrofit2.http.Query("lastName") String lastName, @retrofit2.http.Query("offset") Integer offset, @retrofit2.http.Query("status") String status, @retrofit2.http.Query("limit") Integer limit, @retrofit2.http.Query("orderBy") String orderBy, @retrofit2.http.Query("sortOrder") String sortOrder
  );

  /**
   * List Client Charges
   * The list capability of client charges supports pagination.  Example Requests:  self/clients/1/charges  self/clients/1/charges?offset&#x3D;0&amp;limit&#x3D;5
   * @param clientId clientId (required)
   * @param chargeStatus chargeStatus (optional, default to &quot;all&quot;)
   * @param pendingPayment pendingPayment (optional)
   * @param limit limit (optional)
   * @param offset offset (optional)
   * @return Call&lt;GetSelfClientsClientIdChargesResponse&gt;
   */
  @GET("self/clients/{clientId}/charges")
  Call<GetSelfClientsClientIdChargesResponse> retrieveAllClientCharges1(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Query("chargeStatus") String chargeStatus, @retrofit2.http.Query("pendingPayment") Boolean pendingPayment, @retrofit2.http.Query("limit") Integer limit, @retrofit2.http.Query("offset") Integer offset
  );

  /**
   * List Client Transactions
   * The list capability of client transaction can support pagination.  Example Requests:  self/clients/189/transactions  self/clients/189/transactions?offset&#x3D;10&amp;limit&#x3D;50
   * @param clientId clientId (required)
   * @param offset offset (optional)
   * @param limit limit (optional)
   * @return Call&lt;GetSelfClientsClientIdTransactionsResponse&gt;
   */
  @GET("self/clients/{clientId}/transactions")
  Call<GetSelfClientsClientIdTransactionsResponse> retrieveAllClientTransactions1(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Query("offset") Integer offset, @retrofit2.http.Query("limit") Integer limit
  );

  /**
   * Retrieve client accounts overview
   * An example of how a loan portfolio summary can be provided. This is requested in a specific use case of the community application. It is quite reasonable to add resources like this to simplify User Interface development.  Example Requests:  self/clients/1/accounts   self/clients/1/accounts?fields&#x3D;loanAccounts,savingsAccounts
   * @param clientId clientId (required)
   * @return Call&lt;GetSelfClientsClientIdAccountsResponse&gt;
   */
  @GET("self/clients/{clientId}/accounts")
  Call<GetSelfClientsClientIdAccountsResponse> retrieveAssociatedAccounts1(
    @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * Retrieve a Client Charge
   * Retrieves a Client Charge  Example Requests:  self/clients/1/charges/1   self/clients/1/charges/1?fields&#x3D;name,id
   * @param clientId clientId (required)
   * @param chargeId chargeId (required)
   * @return Call&lt;GetSelfClientsClientIdChargesChargeIdResponse&gt;
   */
  @GET("self/clients/{clientId}/charges/{chargeId}")
  Call<GetSelfClientsClientIdChargesChargeIdResponse> retrieveClientCharge1(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Path("chargeId") Long chargeId
  );

  /**
   * Retrieve a Client Transaction
   * Retrieves a Client TransactionExample Requests:  self/clients/1/transactions/1   self/clients/1/transactions/1?fields&#x3D;id,officeName
   * @param clientId clientId (required)
   * @param transactionId transactionId (required)
   * @return Call&lt;GetSelfClientsClientIdTransactionsTransactionIdResponse&gt;
   */
  @GET("self/clients/{clientId}/transactions/{transactionId}")
  Call<GetSelfClientsClientIdTransactionsTransactionIdResponse> retrieveClientTransaction1(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Path("transactionId") Long transactionId
  );

  /**
   * Retrieve Client Image
   * Optional arguments are identical to those of Get Image associated with an Entity (Binary file)  Example Requests:  self/clients/1/images
   * @param clientId clientId (required)
   * @param maxWidth  (optional)
   * @param maxHeight  (optional)
   * @param output  (optional)
   * @return Call&lt;Void&gt;
   */
  @GET("self/clients/{clientId}/images")
  Call<Void> retrieveImage1(
    @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Query("maxWidth") Integer maxWidth, @retrofit2.http.Query("maxHeight") Integer maxHeight, @retrofit2.http.Query("output") String output
  );

  /**
   * 
   * 
   * @param clientId  (required)
   * @return Call&lt;String&gt;
   */
  @GET("self/clients/{clientId}/obligeedetails")
  Call<String> retrieveObligeeDetails1(
    @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * Retrieve a Client
   * Retrieves a Client  Example Requests:  self/clients/1  self/clients/1?fields&#x3D;id,displayName,officeName
   * @param clientId clientId (required)
   * @return Call&lt;GetSelfClientsClientIdResponse&gt;
   */
  @GET("self/clients/{clientId}")
  Call<GetSelfClientsClientIdResponse> retrieveOne26(
    @retrofit2.http.Path("clientId") Long clientId
  );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfClientApi.kt
package org.openapitools.client.apis

import okhttp3.MultipartBody
import org.openapitools.client.models.GetSelfClientsClientIdAccountsResponse
import org.openapitools.client.models.GetSelfClientsClientIdChargesChargeIdResponse
import org.openapitools.client.models.GetSelfClientsClientIdChargesResponse
import org.openapitools.client.models.GetSelfClientsClientIdResponse
import org.openapitools.client.models.GetSelfClientsClientIdTransactionsResponse
import org.openapitools.client.models.GetSelfClientsClientIdTransactionsTransactionIdResponse
import org.openapitools.client.models.GetSelfClientsResponse
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part
import retrofit2.http.Path
import retrofit2.http.Query

interface SelfClientApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @param contentLength  (optional)
     * @param dateFormat  (optional)
     * @param locale  (optional)
     * @param uploadedInputStream  (optional)
     * @return [kotlin.String]
     */
    @Multipart
    @POST("v1/self/clients/{clientId}/images")
    suspend fun addNewClientImage2(
        @Path("clientId") clientId: Long,
        @Header("Content-Length") contentLength: Long? = null,
        @Part("dateFormat") dateFormat: String? = null,
        @Part("locale") locale: String? = null,
        @Part uploadedInputStream: MultipartBody.Part? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @return [kotlin.String]
     */
    @DELETE("v1/self/clients/{clientId}/images")
    suspend fun deleteClientImage1(@Path("clientId") clientId: Long): String

    /**
     * List Clients associated to the user
     * The list capability of clients can support pagination and sorting.  Example Requests:  self/clients  self/clients?fields&#x3D;displayName,officeName  self/clients?offset&#x3D;10&amp;limit&#x3D;50  self/clients?orderBy&#x3D;displayName&amp;sortOrder&#x3D;DESC
     * Responses:
     *  - 200: OK
     *
     * @param displayName displayName (optional)
     * @param firstName firstName (optional)
     * @param lastName lastName (optional)
     * @param offset offset (optional)
     * @param status status (optional)
     * @param limit limit (optional)
     * @param orderBy orderBy (optional)
     * @param sortOrder sortOrder (optional)
     * @return [GetSelfClientsResponse]
     */
    @GET("v1/self/clients")
    suspend fun retrieveAll36(
        @Query("displayName") displayName: String? = null,
        @Query("firstName") firstName: String? = null,
        @Query("lastName") lastName: String? = null,
        @Query("offset") offset: Int? = null,
        @Query("status") status: String? = null,
        @Query("limit") limit: Int? = null,
        @Query("orderBy") orderBy: String? = null,
        @Query("sortOrder") sortOrder: String? = null
    ): GetSelfClientsResponse

    /**
     * List Client Charges
     * The list capability of client charges supports pagination.  Example Requests:  self/clients/1/charges  self/clients/1/charges?offset&#x3D;0&amp;limit&#x3D;5
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @param chargeStatus chargeStatus (optional, default to "all")
     * @param pendingPayment pendingPayment (optional)
     * @param limit limit (optional)
     * @param offset offset (optional)
     * @return [GetSelfClientsClientIdChargesResponse]
     */
    @GET("v1/self/clients/{clientId}/charges")
    suspend fun retrieveAllClientCharges1(
        @Path("clientId") clientId: Long,
        @Query("chargeStatus") chargeStatus: String? = "all",
        @Query("pendingPayment") pendingPayment: Boolean? = null,
        @Query("limit") limit: Int? = null,
        @Query("offset") offset: Int? = null
    ): GetSelfClientsClientIdChargesResponse

    /**
     * List Client Transactions
     * The list capability of client transaction can support pagination.  Example Requests:  self/clients/189/transactions  self/clients/189/transactions?offset&#x3D;10&amp;limit&#x3D;50
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @param offset offset (optional)
     * @param limit limit (optional)
     * @return [GetSelfClientsClientIdTransactionsResponse]
     */
    @GET("v1/self/clients/{clientId}/transactions")
    suspend fun retrieveAllClientTransactions2(
        @Path("clientId") clientId: Long,
        @Query("offset") offset: Int? = null,
        @Query("limit") limit: Int? = null
    ): GetSelfClientsClientIdTransactionsResponse

    /**
     * Retrieve client accounts overview
     * An example of how a loan portfolio summary can be provided. This is requested in a specific use case of the community application. It is quite reasonable to add resources like this to simplify User Interface development.  Example Requests:  self/clients/1/accounts   self/clients/1/accounts?fields&#x3D;loanAccounts,savingsAccounts
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @return [GetSelfClientsClientIdAccountsResponse]
     */
    @GET("v1/self/clients/{clientId}/accounts")
    suspend fun retrieveAssociatedAccounts2(@Path("clientId") clientId: Long): GetSelfClientsClientIdAccountsResponse

    /**
     * Retrieve a Client Charge
     * Retrieves a Client Charge  Example Requests:  self/clients/1/charges/1   self/clients/1/charges/1?fields&#x3D;name,id
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @param chargeId chargeId
     * @return [GetSelfClientsClientIdChargesChargeIdResponse]
     */
    @GET("v1/self/clients/{clientId}/charges/{chargeId}")
    suspend fun retrieveClientCharge1(
        @Path("clientId") clientId: Long,
        @Path("chargeId") chargeId: Long
    ): GetSelfClientsClientIdChargesChargeIdResponse

    /**
     * Retrieve a Client Transaction
     * Retrieves a Client TransactionExample Requests:  self/clients/1/transactions/1   self/clients/1/transactions/1?fields&#x3D;id,officeName
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @param transactionId transactionId
     * @return [GetSelfClientsClientIdTransactionsTransactionIdResponse]
     */
    @GET("v1/self/clients/{clientId}/transactions/{transactionId}")
    suspend fun retrieveClientTransaction4(
        @Path("clientId") clientId: Long,
        @Path("transactionId") transactionId: Long
    ): GetSelfClientsClientIdTransactionsTransactionIdResponse

    /**
     * Retrieve Client Image
     * Optional arguments are identical to those of Get Image associated with an Entity (Binary file)  Example Requests:  self/clients/1/images
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @param maxWidth  (optional)
     * @param maxHeight  (optional)
     * @param output  (optional)
     * @return [Unit]
     */
    @GET("v1/self/clients/{clientId}/images")
    suspend fun retrieveImage1(
        @Path("clientId") clientId: Long,
        @Query("maxWidth") maxWidth: Int? = null,
        @Query("maxHeight") maxHeight: Int? = null,
        @Query("output") output: String? = null
    ): Unit

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId
     * @return [kotlin.String]
     */
    @GET("v1/self/clients/{clientId}/obligeedetails")
    suspend fun retrieveObligeeDetails2(@Path("clientId") clientId: Long): String

    /**
     * Retrieve a Client
     * Retrieves a Client  Example Requests:  self/clients/1  self/clients/1?fields&#x3D;id,displayName,officeName
     * Responses:
     *  - 200: OK
     *
     * @param clientId clientId
     * @return [GetSelfClientsClientIdResponse]
     */
    @GET("v1/self/clients/{clientId}")
    suspend fun retrieveOne28(@Path("clientId") clientId: Long): GetSelfClientsClientIdResponse

}
